apps/yapms/src/routes/app/[country]/[map]/[year]/initialize/LoadRegions.ts

```diff
@@ -114,6 +116,12 @@ function loadRegions(node: HTMLDivElement): void {
 		console.error('Error Parsing Candidate Data from Map:\n\n' + error);
 	}
 
+	//If map being loaded has the default-mode property set, change the mode.
+	const defaultMode = node.querySelector('svg')?.getAttribute('default-mode');
+	if (ModeSchema.parse(defaultMode)) {
+		ModeStore.set(defaultMode as Mode);
+	}
```

Comment:
With Zod we don't need to cast any data.

We can either use parse or safe parse, and use type narrowing.

parse will throw an error you can catch
```ts
try {
  const defaultModeAttribute = node.querySelector('svg')?.getAttribute('default-mode');
  const defaultMode = ModeSchema.parse(defaultModeAttribute);
  ModeStore.set(defaultMode);
} catch (error) {
  console.error('Error Parsing Mode Data from Map:\n\n' + error);
}
```

safeParse will return an object with with data about if it's an error or not.
```ts
const defaultModeAttribute = node.querySelector('svg')?.getAttribute('default-mode');
const defaultMode = ModeSchema.safeParse(defaultModeAttribute);
if (defaultMode.success) {
  ModeStore.set(defaultMode.data);
}
```

I think whichever one you think is cleanest will be fine here.

Reply:
By having the parse in the function, it will cause the loadRegions function to throw an error if it finds an invalid default mode. It should default to something if it doesn't load proper data.

Reply:
Right good, catch. Will change to safeParse.